### PR TITLE
Add mypy and more typing

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Please check the `item_id` typing. It seems it should be an integer by default, but sometimes we need to copy it to string, eg when using it for the topic.
- Also check the `OZWManager.EVENT_CHANGED` attribute. It was `None`, but I wanted to avoid making the `event` parameter of `OZWOptions.notify` optional so I changed it to a placeholder string. Not sure if my solution is the best.

### CODE REVIEW

File: openzwavemqtt/base.py [hunk 1]
```diff
@@ -39,11 +39,12 @@ def process_message(self, topic: Deque[str], message: dict):
             return
 
         if item is None:
-            topic_part = item_id
+            topic_part = str(item_id)
```
Comment (MartinHjelmare) [inline openzwavemqtt/base.py:42]: Note this copy to string.

File: openzwavemqtt/manager.py [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+
 
 class OZWManager(ZWaveBase):
     """Manager that holds the OZW instances connected to MQTT."""
 
     DIRECT_COLLECTION = "instance"
-    DEFAULT_VALUE = None
-    EVENT_CHANGED = None
+    DEFAULT_VALUE: Optional[dict] = None
+    EVENT_CHANGED = "manager_placeholder_event"
```
Comment (MartinHjelmare) [inline openzwavemqtt/manager.py:20]: Here I've change None for a string.

**Comment by MartinHjelmare** [comment]
@balloob please have a look.

File: requirements_dev.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+-r requirements.txt
```
Comment (MartinHjelmare) [inline requirements_dev.txt:1]:
```suggestion
```

File: openzwavemqtt/base.py (hunk 1 above)
Comment (balloob) [reply openzwavemqtt/base.py:42]: Surprised this never failed in a test?

File: tox.ini [hunk 4]
```diff
@@ -1,10 +1,10 @@
 [tox]
-envlist = py36, py37, py38, lint
+envlist = py36, py37, py38, lint, mypy
 skip_missing_interpreters = True
 
 [gh-actions]
 python =
-  3.6: py36, lint
+  3.6: py36, lint, mypy
```
Comment (balloob) [inline tox.ini:7]: We should consider dropping py36.

Comment (MartinHjelmare) [reply tox.ini:7]: It's 1.5 years until EOL for Python 3.6. I don't mind dropping it, but should we make a general policy? Should we say that we support the two latest Python versions?

Comment (balloob) [reply tox.ini:7]: Yeah. Follow HA :)

Comment (MartinHjelmare) [reply tox.ini:7]: Ok, I'll drop 3.6 and document this in a separate PR.

**Pull request merged**